server/pkg/slice/strings.go

```diff
@@ -1,15 +1,15 @@
 package slice
 
-func ContainsAny[C comparable](haystack []C, needles ...C) bool {
-	for _, h := range haystack {
+func ContainsAny[C comparable](haystack []C, needles ...C) (bool, int) {
```

Comment:
same comment as with the uint64 slice.
you could also consider replacing the contents of `HasUint64` with a call to this, since it's a generic.